src/js/11-elastic-search.js

```diff
@@ -0,0 +1,125 @@
+        this.searchresults = ''
+        resultList.results.documents.forEach((result) => {
+          this.searchresults += '<a href="search.html?query=' + encodeURIComponent(result.suggestion) + '">' + result.suggestion + '</a>'
+        })
+        if (this.searchresults) {
+          document.getElementById('search-results').innerHTML = '<div id="the-results">' + this.searchresults + '</div>'
+        } else {
+          document.getElementById('search-results').innerHTML = ''
+        }
+      })
+      .catch((error) => {
+        console.log(`error: ${error}`)
+      })
+  }
+
+  getResults (searchKey) {
+    this.client
+      .search(searchKey, this.options)
+      .then((resultList) => {
+        this.searchresults = ''
+        this.createPagination(resultList.info.meta.page.current, resultList.info.meta.page.total_pages)
+        document.getElementById('searchnores').innerHTML = resultList.info.meta.page.total_results
+        resultList.results.forEach((result) => {
+          this.searchresults += '<a class="the-search-result" href="' + result.data.url.raw + '"><span class="result-title">' + result.data.title.raw + '</span><span class="result-description">' + result.data.headings.raw + '</span><span class="result-url">' + result.data.url.raw + '</span></a>'
+        })
+        document.getElementById('search-page-results').innerHTML = this.searchresults
+      })
+      .catch((error) => {
+        console.log(`error: ${error}`)
+      })
+  }
+
+  createPagination (pageId, totalPages) {
+    const urlResult = decodeURI(window.location.search.split('?query=')[1].split('&page=')[0])
+    const newUrl = '?query=' + encodeURIComponent(urlResult)
+    const previousPageNumber = pageId - 1
+    const nextPageNumber = pageId + 1
+    let thePages = ''
+    if (pageId > 1) {
+      thePages += '<a href="' + newUrl + '&page=' + previousPageNumber + '" class="previous"><i class="fa fa-angle-left"></i></a>'
+      thePages += '<a href="' + newUrl + '&page=' + previousPageNumber + '">' + previousPageNumber + '</a>'
+    }
+    thePages += '<a href="' + newUrl + '&page=' + pageId + '" class="active">' + pageId + '</a>'
+    if (pageId < totalPages) {
+      thePages += '<a href="' + newUrl + '&page=' + nextPageNumber + '">' + nextPageNumber + '</a>'
+      thePages += '<a href="' + newUrl + '&page=' + nextPageNumber + '" class="next"><i class="fa fa-angle-right"></i></a>'
+    }
+    document.getElementById('search-page-pagination').innerHTML = thePages
+  }
+}
+
+(function () {
+  $(document).ready(function () {
+    let timeout = false
+    const elasticSearch = new ElasticSearch()
+    const searchPageResults = document.getElementById('search-page-results')
+    const searchIcon = document.getElementById('search-icon')
+    const searchBar = document.getElementById('searchbar')
+    const searchText = document.getElementById('search-input')
+    const currentLang = window.location.href.includes('/en-gb/') ? 'en-gb' : 'de-de'
```

Comment:
```suggestion
    const currentLang = window.location.href.includes('/en-gb/') ? 'en' : 'de'
```
Not sure about this, but the [language optimisation guide](https://www.elastic.co/guide/en/app-search/7.17/language-optimization-guide.html) doesn't seem to include the locale, so maybe we can't do that either?